### PR TITLE
dataflow: describe error handling approach

### PR DESCRIPTION
**Pull request by benesch**
Requested by @frankmcsherry in #2441. @frankmcsherry, how's this?

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/2640)

### CODE REVIEW

File: src/dataflow/render/mod.rs [hunk 1]
```diff
@@ -7,6 +7,95 @@
...
+//!     oks   errs
+//! ```
+//!
+//! The project operation cannot fail, so errors from errs1 are propagated
+//! directly. Map operators are fallible and so can inject additional errors
+//! into the stream. Join operators combine the errors from each of their
+//! inputs.
+//!
+//! The semantics of the error stream are minimal. From the perspective of SQL,
+//! a dataflow is considered to be in an error state if there is more than one
```
Comment (frankmcsherry) [inline src/dataflow/render/mod.rs:61]: More than zero?

Comment (benesch) [reply src/dataflow/render/mod.rs:61]: Whoops, thanks!

**Comment by benesch** [comment]
Buglet fixed! Thanks for catching that. Does it seem good otherwise?

**Comment by frankmcsherry** [comment]
Yeah, looks good to me! What is up with the test?

**Comment by benesch** [comment]
Eep, seems like the test is racy. Unrelated to this PR, though.

**Pull request merged**